ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -312,6 +312,67 @@ public QueryMetadata getQueryExecutionPlan(final Query query, final KsqlConfig k
     }
   }
 
+  public List<Pair<String, Statement>> parseStatements(
```

Comment:
How is this function different from parseQueries? They seem to be doing the same thing. If there is some difference, can we try and share some of the code?

Reply:
There are some minor differences. The list of returned values are not the same since `parseQueries` only returns the values that have non null values returned from `buildSingleQueryAst`.
We can merge the two functions but it would need some changes on other parts including adding `UNSET` to DDL statements. This can be done in a separate PR.

Reply:
IMHO this should be done in this PR. 

In this PR you are introducing a new method, that is very similar to an existing method. Much better to do the refactor needed now. As saying 'this can be done in a separate PR' likely means 'this will never be done'.